File: Leetcode/0701/701_Insert_into_a_Binary_Search_Tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} val
 * @return {TreeNode}
 */
var insertIntoBST = function(root, val) {
  helper(root, val);
  return root;
};

var helper = function(root, val) {
  console.log("root is ", root);
  if (root === null) {
    return new TreeNode(val);
  }
  if (root.val > val) {
    root.left = helper(root.left, val);
    return root;
  }
  root.right = helper(root.right, val);
  return root;
};

File: Leetcode/0701/701_Insert_into_a_Binary_Search_Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    TreeNode* insertIntoBST(TreeNode* root, int val) {
        if(root == nullptr)
        {
            return new TreeNode(val);
        }
        
        if(root->val < val)
        {
            root->right = insertIntoBST(root->right, val);
        }
        else
        {
           root->left = insertIntoBST(root->left, val);  
        }
        
        return root;
    }
};